dot/state/block_funcs_test.go

```diff
@@ -0,0 +1,182 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+import (
+	"github.com/ChainSafe/gossamer/dot/core"
+	"github.com/ChainSafe/gossamer/dot/core/types"
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/common/optional"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/tests"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+// testGenesisHeader is a test block header
+var testGenesisHeader = &types.Header{
+	Number:    big.NewInt(0),
+	StateRoot: trie.EmptyHash,
+}
+
+// newTestCoreService creates a new test core service
+func newTestCoreService(t *testing.T, cfg *core.Config) *core.Service {
+	if cfg == nil {
+		rt := runtime.NewTestRuntime(t, tests.POLKADOT_RUNTIME)
+		cfg = &core.Config{
+			Runtime:         rt,
+			IsBabeAuthority: false,
+		}
+	}
+
+	if cfg.Keystore == nil {
+		cfg.Keystore = keystore.NewKeystore()
+	}
+
+	if cfg.NewBlocks == nil {
+		cfg.NewBlocks = make(chan types.Block)
+	}
+
+	if cfg.MsgRec == nil {
+		cfg.MsgRec = make(chan network.Message, 10)
+	}
+
+	if cfg.MsgSend == nil {
+		cfg.MsgSend = make(chan network.Message, 10)
+	}
+
+	if cfg.SyncChan == nil {
+		cfg.SyncChan = make(chan *big.Int, 10)
+	}
+
+	stateSrvc := NewService("")
+	stateSrvc.UseMemDB()
+
+	err := stateSrvc.Initialize(testGenesisHeader, trie.NewEmptyTrie(nil))
+	require.Nil(t, err)
+
+	err = stateSrvc.Start()
+	require.Nil(t, err)
+
+	if cfg.BlockState == nil {
+		cfg.BlockState = stateSrvc.Block
+	}
+
+	if cfg.StorageState == nil {
+		cfg.StorageState = stateSrvc.Storage
+	}
+
+	s, err := core.NewService(cfg)
+	require.Nil(t, err)
+
+	return s
+}
+
+func TestGetSetReceiptMessageQueueJustification(t *testing.T) {
+	tt := trie.NewEmptyTrie(nil)
+	rt := runtime.NewTestRuntimeWithTrie(t, tests.POLKADOT_RUNTIME, tt)
+
+	kp, err := sr25519.GenerateKeypair()
+	require.Nil(t, err)
+
+	pubkey := kp.Public().Encode()
+	err = tt.Put(tests.AuthorityDataKey, append([]byte{4}, pubkey...))
+	require.Nil(t, err)
+
+	ks := keystore.NewKeystore()
+	ks.Insert(kp)
+
+	cfg := &core.Config{
+		Runtime:         rt,
+		Keystore:        ks,
+		IsBabeAuthority: false,
+	}
+
+	s := newTestCoreService(t, cfg)
+	require.NotNil(t, s)
```

Comment:
s isn't used anywhere past this line, also state shouldn't be importing core, I don't see why core is needed at all here. you can make a blockState with `NewBlockState` or just `&BlockState{}`

Reply:
correct, that avoids all the boiler plate code, thanks for the hint! :)